paddle/operators/softmax_op.cc

```diff
@@ -18,14 +18,13 @@ namespace operators {
 
 class SoftmaxOp : public OperatorWithKernel {
 protected:
-  void InferShape(const std::vector<const Tensor *> &inputs,
-                  const std::vector<Tensor *> &outputs) const override {
-    PADDLE_ENFORCE(inputs.size() == 1, "Only one input is need for softmax");
-    PADDLE_ENFORCE(inputs[0]->dims().size() == 2,
+  void InferShape(const InferShapeContext &ctx) const override {
+    PADDLE_ENFORCE(ctx.InputSize() == 1, "Only one input is need for softmax");
```

Comment:
same as top, when an operator InferShape, it means to check the weather the real inputs and outputs match the definition in its op_proto.

here the `1` is `softmax_op.op_proto.inputs().size()`, and `OperatorBase` can do this enforce automatically for `SoftmaxOp`.